crates/networking/p2p/sync.rs

```diff
@@ -122,9 +122,9 @@ impl SyncManager {
             trie_rebuilder: None,
             // This won't be used
             cancel_token: CancellationToken::new(),
-            blockchain: Blockchain::default_with_store(
+            blockchain: Arc::new(Blockchain::default_with_store(
```

Comment:
what if `Blockchain::default_with_store` already returns an `Arc`? Would that make sense?

Reply:
Just asking, not sure it's the right approach.